evalml/pipelines/component_graph.py

```diff
@@ -414,7 +421,10 @@ def _transform_features(
             self.input_feature_names.update({component_name: list(x_inputs.columns)})
             if isinstance(component_instance, Transformer):
                 if fit:
-                    output = component_instance.fit_transform(x_inputs, y_input)
+                    if component_instance._is_fitted and self.cached_data is not None:
```

Comment:
In what case is the component_instance fitted but cached_data is None? I guess I'm wondering why this isn't just checking `_is_fitted`

Reply:
Good point! I think I was worried since I no longer pass in newly-initialized pipelines for the stacked ensemble pipeline, but since I'm grabbing the component from the cache only after checking if the cache exists, then this check is repetitive. Removing!